### PR TITLE
Add CRUD actions for Schema Registry service

### PR DESCRIPTION
**Pull request by IldarAlmakaev**
The following endpoints have been implemented:
**Get all schema names**
``GET /api/clusters/{clusterName}/schemas``

**Get all schema versions**
`GET /api/clusters/{clusterName}/schemas/{schemaName}/versions`

**Get/Delete the latest schema version**
`GET/DELETE /api/clusters/{clusterName}/schemas/{schemaName}/latest`

**Get/Delete schema by specified version**
`GET/DELETE /api/clusters/{clusterName}/schemas/{schemaName}/versions/{version}`

**Create a new schema or delete a schema entirely**
`POST/DELETE` /api/clusters/{clusterName}/schemas/{schemaName}

**Get/Update schema compatibility on a schema level** 
`GET/PUT /api/clusters/{clusterName}/schemas/compatibility/{schemaName}`

**Get/Update schema compatibility on global level**
`GET/PUT /api/clusters/{clusterName}/schemas/compatibility`

**Check a schema compatibility with its latest version**
`POST /api/clusters/{clusterName}/schemas/compatibility/{schemaName}/check`

### CODE REVIEW

**Review by workshur — APPROVED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/SchemaRegistryService.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    public Flux<SchemaSubject> getSchemaSubjectByVersion(String clusterName, String subjectName, Integer version) {
+        return this.getSchemaSubject(clusterName, subjectName, String.valueOf(version));
+    }
+
+    public Flux<SchemaSubject> getLatestSchemaSubject(String clusterName, String subjectName) {
+        return this.getSchemaSubject(clusterName, subjectName, LATEST);
+    }
+
+    private Flux<SchemaSubject> getSchemaSubject(String clusterName, String subjectName, String version) {
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/SchemaRegistryService.java:65]: Maybe it should be Mono?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/SchemaRegistryService.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    public Mono<ResponseEntity<Void>> deleteLatestSchemaSubject(String clusterName, String subjectName) {
+        return this.deleteSchemaSubject(clusterName, subjectName, LATEST);
+    }
+
+    private Mono<ResponseEntity<Void>> deleteSchemaSubject(String clusterName, String subjectName, String version) {
+        return clustersStorage.getClusterByName(clusterName)
+                .map(cluster -> webClient.delete()
+                        .uri(cluster.getSchemaRegistry() + URL_SUBJECT_BY_VERSION, subjectName, version)
+                        .retrieve()
+                        .onStatus(HttpStatus.NOT_FOUND::equals, resp -> Mono.error(new NotFoundException("No such subject or version")))
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/SchemaRegistryService.java:88]: 1. What about 5xx statuses
2. Don't you mind make status handling generic?

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 3]
```diff
@@ -335,6 +335,339 @@ paths:
...
+            type: string
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/SchemaSubject'
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:465]: Why is it array?

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 4]
```diff
@@ -335,6 +335,339 @@ paths:
...
+          required: true
+          schema:
+            type: integer
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:517]: Why is it array?

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 5]
```diff
@@ -335,6 +335,339 @@ paths:
...
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CompatibilityLevel'
+      responses:
+        200:
+          description: OK
+        404:
+          description: Not Found
+
+  /api/clusters/{clusterName}/schemas/compatibility/{schemaName}:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:588]: let's return compatibility inside schemaobject

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 6]
```diff
@@ -335,6 +335,339 @@ paths:
...
+          schema:
+            type: string
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CompatibilityLevelResponse'
+    put:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:612]: /api/clusters/{clusterName}/schemas/compatibility/{schemaName} ->
PUT /api/clusters/{clusterName}/schemas/{schemaName}/compatibility

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 7]
```diff
@@ -335,6 +335,339 @@ paths:
...
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CompatibilityLevel'
+      responses:
+        200:
+          description: OK
+        404:
+          description: Not Found
+
+  /api/clusters/{clusterName}/schemas/compatibility/{schemaName}/check:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:639]: /api/clusters/{clusterName}/schemas/{schemaName}/check

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 8]
```diff
@@ -657,4 +990,58 @@ components:
...
+    SchemaSubject:
+      type: object
+      properties:
+        subject:
+          type: string
+        version:
+          type: string
+        id:
+          type: integer
+        schema:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:1004]: Let's add compatibility here

**Review by germanosin — CHANGES_REQUESTED** [review]

**Review by germanosin — APPROVED** [review]

**Pull request merged**